### ISSUE TITLE
IDISP013 False Positive?

### ISSUE THREAD

**Issue by Bouke** [DotNetAnalyzers/IDisposableAnalyzers#581]
IDISP013 triggers in the following situation:

```csharp
private async Task IDISP013()
{
    using var client = new HttpClient();

    Func<Task<HttpResponseMessage>> f = () => client.GetAsync("");
    //                                        ^^^^^^^^^^^^^^^^^^^ Await in using

    await f();
}
```

I believe this to be a false positive, as `f` is executed within the scope of the `using`. -- When `f` wasn't awaited I can see there being an issue where `client` would've been disposed before `GetAsync` finished.

**Comment by dlebansais** [comment]
To be sure there is no error to report, they would need to be certain that `f` is awaited everywhere in the method, which can be a problem if the code becomes complicated. Consider:

```cs
private async Task IDISP013()
{
    using var client = new HttpClient();

    Func<Task<HttpResponseMessage>> f = () => client.GetAsync("");
 
    if (SomeTest)
    {
        await f();
    }
    else
    {
        AwaitLater(f);
    }
}
```

I tried to solve your error, this is how I did it:

```cs
static async Task IDISP013(Uri uri)
{
    using var client = new HttpClient();

    Func<Task<HttpResponseMessage>> f = async () => await client.GetAsync(uri).ConfigureAwait(false);

    using var msg = await f().ConfigureAwait(false);
}
```

Does it work for you?
